[esx]/es_extended/server/classes/player.lua

```diff
@@ -40,10 +45,23 @@ function CreateExtendedPlayer(playerId, identifier, group, accounts, inventory,
 	end
 
 	function self.updateCoords()
-		local Ped = GetPlayerPed(self.source)
-		local coords = GetEntityCoords(Ped)
-		local heading = GetEntityHeading(Ped)
-		self.coords = {x = ESX.Math.Round(coords.x, 1), y = ESX.Math.Round(coords.y, 1), z = ESX.Math.Round(coords.z, 1), heading = ESX.Math.Round(heading or 0.0, 1)}
+		SetTimeout(1000,function()
+			local Ped = GetPlayerPed(self.source)
+			if DoesEntityExist(Ped) then
+				local coords = GetEntityCoords(Ped)
+				local distance = #(coords - vector3(self.coords.x, self.coords.y, self.coords.z))
+				if distance > 1.5 then
+					local heading = GetEntityHeading(Ped)
+					self.coords = {
+						x = coords.x,
+						y = coords.y, 
+						z = coords.z, 
+						heading = heading or 0.0
+					}
+				end
+			end
+			self.updateCoords()
+		end)
```

Comment:
this loop continues even though the player had already disconnected, doesn't it?

Reply:
Yes, I also think that the timer will not stop.

Reply:
> this loop continues even though the player had already disconnected, doesn't it?

no, since its within the xPlayer class, which only exists when the player is connected, once they use the relog function or disconnect, the function will no longer exist, and therefore, not continue.

Reply:
if it uses a thread, that would be the case though, which is why it's a threadless design 🙃

Reply:
I had tested it using print, and unfortunately it still printing even I had already disconnected.

Reply:
that literally isn't possible 👀, it doesn't exist after you disconnect

Reply:
I believe SetTimeout creates a thread internally which would lead to having a lot of threads the more players' count increases. I could be wrong tho...

Reply:
> I believe SetTimeout creates a thread internally which would lead to having a lot of threads the more players' count increases if using this method...

as far as i can tell from looking internally, that is not the case :)

Reply:
> > I believe SetTimeout creates a thread internally which would lead to having a lot of threads the more players' count increases if using this method...
> 
> as far as i can tell from looking internally, that is not the case :)

As far as I can tell that is the case. check [here](https://github.com/citizenfx/fivem/blob/master/code/components/citizen-scripting-lua/src/LuaScriptRuntime.cpp#L1032-L1144)